File: app/src/main/java/com/lazycouple/restapiclient/util/BindingHolder.java
package com.lazycouple.restapiclient.util;

import android.databinding.ViewDataBinding;
import android.support.v7.widget.RecyclerView;



public abstract class BindingHolder<T> extends RecyclerView.ViewHolder {
    protected final ViewDataBinding binding;

    public BindingHolder(ViewDataBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public abstract void bind(T variable);

    public ViewDataBinding getBinding() {
        return binding;
    }
}

File: app/src/main/java/com/lazycouple/restapiclient/util/BindingHolder.kt
package com.lazycouple.restapiclient.util

import android.databinding.ViewDataBinding
import android.support.v7.widget.RecyclerView


abstract class BindingHolder<T>(val binding: ViewDataBinding) : RecyclerView.ViewHolder(binding.root) {

    abstract fun bind(variable: T)
}
